lib/exercises.rb

```diff
@@ -22,8 +71,51 @@ def top_k_frequent_elements(list, k)
 #   Each element can either be a ".", or a digit 1-9
 #   The same digit cannot appear twice or more in the same 
 #   row, column or 3x3 subgrid
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: I want to say this is O(1) since the number of elements in a 3x3 grid
+# you have to check is always the same, and any lookups and inserts are O(1) since I'm not 
+# only storing everything in a hash, but I'm mapping the keys to another hash. 
+
+# Space Complexity: I also think if we are always using a 3x3 grid and inserting into a hash, 
+# this should technically be O(1) since all of the hashes will only expand to the number of 
+# elements in a 3x3 grid, which doesn't change, or would only increase by a constant.
 def valid_sudoku(table)
```

Comment:
You are correct that this is O(1) because Sudoku is always a 9x9 grid and never grows!